### PR TITLE
🎨 adding `folder_id` to project resource

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🎨 adding `folder_id` to project resource (This folder ID depends on the context of the user retrieving the project, as the user may have the shared project in their private folder.) 

## Related issue/s
- relates https://github.com/ITISFoundation/osparc-issues/issues/716

## How to test
- both private and shared workspace project listing tested :heavy_check_mark: 

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]
thanks!

File: services/web/server/tests/unit/with_dbs/02/test_projects_states_handlers.py [hunk 1]
```diff
@@ -866,10 +866,10 @@ async def test_close_project(
 @pytest.mark.parametrize(
     "user_role, expected",
     [
-        (UserRole.ANONYMOUS, status.HTTP_401_UNAUTHORIZED),
+        # (UserRole.ANONYMOUS, status.HTTP_401_UNAUTHORIZED),
         (UserRole.GUEST, status.HTTP_200_OK),
-        (UserRole.USER, status.HTTP_200_OK),
-        (UserRole.TESTER, status.HTTP_200_OK),
+        # (UserRole.USER, status.HTTP_200_OK),
+        # (UserRole.TESTER, status.HTTP_200_OK),
```
Comment (GitHK) [inline services/web/server/tests/unit/with_dbs/02/test_projects_states_handlers.py:872]: This to me looks bad

File: services/web/server/src/simcore_service_webserver/projects/models.py [hunk 2]
```diff
@@ -78,6 +86,19 @@ class Config:
...
+    workspace_id: WorkspaceID | None  # None if it's a private workspace
+    read: bool
+    write: bool
+    delete: bool
+
+    class Config:
+        orm_mode = True
+
+
+# class UserProjectAccessWithWork
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/models.py:100]: looks like some leftover

File: services/web/server/src/simcore_service_webserver/projects/db.py [hunk 3]
```diff
@@ -496,13 +502,9 @@ async def get_project(
         only_published: bool = False,
         only_templates: bool = False,
     ) -> tuple[ProjectDict, ProjectType]:
-        """Returns all projects *owned* by the user
-
-            - prj_owner
-            - Notice that a user can have access to a template but he might not own it
-            - Notice that a user can have access to a project where he/she has read access
-
-        :raises ProjectNotFoundError: project is not assigned to user
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/db.py:505]: could you please bring this back

File: services/web/server/src/simcore_service_webserver/projects/db.py [hunk 4]
```diff
@@ -496,13 +502,9 @@ async def get_project(
         only_published: bool = False,
         only_templates: bool = False,
     ) -> tuple[ProjectDict, ProjectType]:
-        """Returns all projects *owned* by the user
-
-            - prj_owner
-            - Notice that a user can have access to a template but he might not own it
-            - Notice that a user can have access to a project where he/she has read access
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/projects/db.py:503]: Also why was this removed? The part you added looks like some complementary information

File: services/web/server/tests/unit/with_dbs/02/test_projects_states_handlers.py (hunk 1 above)
Comment (matusdrobuliak66) [reply services/web/server/tests/unit/with_dbs/02/test_projects_states_handlers.py:872]: thanks!

File: services/web/server/src/simcore_service_webserver/projects/models.py (hunk 2 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/models.py:100]: correct!

File: services/web/server/src/simcore_service_webserver/projects/db.py (hunk 3 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/db.py:505]: Those comments were already not valid. User id and access were removed from this function in some previous PR.

File: services/web/server/src/simcore_service_webserver/projects/db.py (hunk 4 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/db.py:503]: the same as above

**Review by GitHK — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/projects/_access_rights_api.py [hunk 5]
```diff
@@ -51,19 +51,31 @@ async def get_user_project_access_rights(
...
+        _user_project_access_rights_with_workspace = (
+            UserProjectAccessRightsWithWorkspace(
+                uid=user_id,
+                workspace_id=None,
+                read=_user_project_access_rights.read,
+                write=_user_project_access_rights.write,
+                delete=_user_project_access_rights.delete,
+            )
+        )
+    return _user_project_access_rights_with_workspace
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_access_rights_api.py:78]: ok.  In this context,  it makes sense indeed why you distinguish between `_user_project_access_rights` and `_user_project_access_rights_with_workspace`. But IMO it is a detail that should not be exposed outside of this function. Therefore I still think the model returned here should basically be a class suggesting "project access rights" without referring to "workspaces"

**Review by pcrespov — APPROVED** [review]
Thx. I left some suggestions.

Summary of this PR: essentially you are now passing `folder_id` as part of the response model. correcty?

File: packages/models-library/src/models_library/api_schemas_webserver/projects.py [hunk 6]
```diff
@@ -84,6 +84,7 @@ class ProjectGet(OutputSchema):
     dev: dict | None
     permalink: ProjectPermalink = FieldNotRequired()
     workspace_id: WorkspaceID | None
+    folder_id: FolderID | None
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply packages/models-library/src/models_library/api_schemas_webserver/projects.py:87]: Yes current we have 2 concepts/resources
 - **workspaces**
   - If NULL it means it is User private workspace
   - If not NULL it is a shared workspace (Yes you can make abstraction that it can be viewed as a mount)
-  **folders**
  -  if NULL it is a root folder
  -  if not NULL it is a normal folder with ID
  - NOTE: Folders can live in both Private but also Shared workspaces

File: services/web/server/src/simcore_service_webserver/projects/_access_rights_api.py (hunk 5 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/_access_rights_api.py:78]: Yes, currently you get access to the project based on different context (private workspace or shared workspace) it is useful to have this information in the output of this function as you can see I use it in some places.

File: services/web/server/src/simcore_service_webserver/projects/models.py [hunk 7]
```diff
@@ -78,6 +86,19 @@ class Config:
         orm_mode = True
 
 
+class UserProjectAccessRightsWithWorkspace(BaseModel):
```
[the remark this answers is not in the document]
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/models.py:89]: Answer bellow
